tests/integration/local/start_api/test_start_api.py

```diff
@@ -1549,20 +1559,92 @@ class TestServiceCorsSwaggerRequestsWithHttpApi(StartApiIntegBaseClass):
     def setUp(self):
         self.url = "http://127.0.0.1:{}".format(self.port)
 
+    def assertCors(self, response):
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.headers.get("Access-Control-Allow-Origin"), "*")
+        self.assertEqual(response.headers.get("Access-Control-Allow-Headers"), "origin")
+        self.assertEqual(response.headers.get("Access-Control-Allow-Methods"), "GET,OPTIONS,POST")
+        self.assertEqual(response.headers.get("Access-Control-Allow-Credentials"), "true")
+        self.assertEqual(response.headers.get("Access-Control-Max-Age"), "42")
+
     @pytest.mark.flaky(reruns=3)
     @pytest.mark.timeout(timeout=600, method="thread")
     def test_cors_swagger_options_httpapi(self):
         """
-        This tests that the Cors are added to option requests in the swagger template
+        This tests that the Cors headers are added to OPTIONS responses
         """
-        response = requests.options(self.url + "/httpapi-echobase64eventbody", timeout=300)
+        response = requests.options(self.url + "/httpapi-echobase64eventbody", headers={"Origin": "https://abc"}, timeout=300)
+        self.assertCors(response)
+
+    def test_cors_swagger_get_httpapi(self):
+        """
+        This tests that the Cors headers are added to _other_ method requests
+        """
+        response = requests.get(self.url + "/httpapi-echobase64eventbody", headers={"Origin": "https://abc"}, timeout=300)
+        self.assertCors(response)
+
+
 
+class TestServiceCorsComplexHttpApi(StartApiIntegBaseClass):
+    """
+    This test covers following cases:
+    - Cors headers added to OPTIONS responses
+    - Cors headers added to other method responses
+    - Cors configuration supports multiple origins, allow headers and methods
+    """
+
+    template_path = "/testdata/start_api/template-cors.yaml"
+    binary_data_file = "testdata/start_api/binarydata.gif"
+
+    def setUp(self):
+        self.url = "http://127.0.0.1:{}/test".format(self.port)
+
+    def assertPresence(self, response):
+        expected_origin = response.request.headers.get("Origin")
```

Comment:
Would it be possible to pass `expected_origin` as a parameter instead of grabbing it from the `response.request` object? My thinking here was that this way, we can pass around concrete values to test against instead of testing the expected origin via the `request` object